packages/amplify_api/lib/src/graphql/graphql_response_decoder.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:convert';
+
+import 'package:amplify_api/amplify_api.dart';
+
+class GraphQLResponseDecoder {
+  // Singleton methods/properties
+  // usage: GraphQLResponseDecoder.instance;
+  GraphQLResponseDecoder._();
+
+  static final GraphQLResponseDecoder _instance = GraphQLResponseDecoder._();
+
+  static GraphQLResponseDecoder get instance => _instance;
+
+  GraphQLResponse<T> decode<T>(
+      {required GraphQLRequest request,
+      required String data,
+      required List<GraphQLResponseError> errors}) {
+    // if no ModelType fallback to default
+    if (request.modelType == null) {
+      return GraphQLResponse<T>(data: data as T, errors: errors);
```

Comment:
This cast will fail for everything but T == String. Probably better to write this piece as

```dart
if (request.modelType == null) {
  if (T == String) {
    return GraphQLResponse(data: data as T, errors: errors); // <T> is implied
  } else {
    // handle case (probably throw)
  }
}
```

Reply:
Added check and threw an error.